src/webgpu/shader/execution/builtin/builtin.ts

```diff
@@ -205,14 +208,20 @@ function toStorage(ty: Type, expr: string): string {
   }
   if (ty instanceof VectorType) {
     if (ty.elementType.kind === 'bool') {
-      return `select(vec${ty.width}<u32>(0u), vec${ty.width}<u32>(0u), ${expr})`;
+      return `select(vec${ty.width}<u32>(0u), vec${ty.width}<u32>(1u), ${expr})`;
```

Comment:
Ah. That explains it.
Sorry I didn't catch this on the earlier review.

Reply:
No need to apologise, my stupid typo.